unified-scheduler-pool/src/lib.rs

```diff
@@ -1804,7 +1845,18 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
                     // session_result_sender just above
                     let mut new_result_with_timings = None;
 
+                    let mut discard_on_reset = false;
                     loop {
+                        if discard_on_reset {
+                            discard_on_reset = false;
+                            let mut count = Saturating(0);
+                            while let Some(task) = state_machine.schedule_next_unblocked_task() {
+                                state_machine.deschedule_task(&task);
+                                count += 1;
+                            }
```

Comment:
is there not a faster way to just clear all outstanding tasks?

Reply:
yeah. i think so. how about this explanation?: https://github.com/anza-xyz/agave/pull/6246/commits/d26a7cbe0f49ef58460374485a308d6d506a78b1